modules/ShowersMatcher.py

```diff
@@ -0,0 +1,427 @@
+    #         C = np.zeros((len(pairs), len(free_nodes)))
+    #         for i, p in enumerate(pairs):
+    #             for j, f in enumerate(free_nodes):
+    #                 score = 0
+    #                 type_f = matched_full_graph.nodes(data=True)[f]['type']
+    #                 idx_f = matched_full_graph.nodes(data=True)[f]['idx']
+    #                 matching_mode = -1
+    #                 # Length of either p[0] will be 1 or p[1] will be 1
+    #                 if len(p[0]) == 1:
+    #                     # A pred shower is matched to one or more truth showers
+    #                     if type_f == 0:
+    #                         matching_mode = 0
+    #                     else:
+    #                         pass
+    #                 if len(p[1]) == 1:
+    #                     # A truth shower is matched to one or more pred showers
+    #                     if type_f == 0:
+    #                         pass
+    #                     else:
+    #                         matching_mode = 1
+    #
+    #                 if matching_mode == 0:
+    #                     sid_p = next(iter(p[0]))
+    #                     idx_p = matched_full_graph.nodes(data=True)[sid_p]['idx']
+    #                     numerator = intersection_matrix[idx_p, idx_f]
+    #                     denominator1 = pred_sum_matrix[idx_p, 0]
+    #                     denominator2 = truth_sum_matrix[0, idx_f]
+    #                     score = numerator / min(denominator1, denominator2)
+    #                 elif matching_mode == 1:
+    #                     sid_t = next(iter(p[1]))
+    #                     idx_t = matched_full_graph.nodes(data=True)[sid_t]['idx']
+    #                     numerator = intersection_matrix[idx_f, idx_t]
+    #                     denominator1 = truth_sum_matrix[0, idx_t]
+    #                     denominator2 = pred_sum_matrix[idx_f, 0]
+    #                     score = numerator / min(denominator1, denominator2)
+    #                 if score > self.iou_threshold:
+    #                     C[i, j] = score
+    #
+    #         # # Let's match these guys again
+    #         C = np.array(C)
+    #         row_id, col_id = linear_sum_assignment(C, maximize=True)
+    #         for r, c in zip(row_id, col_id):
+    #             if C[r, c] > 0:
+    #                 p = pairs[r]
+    #                 if matched_full_graph.nodes(data=True)[free_nodes[c]]['type'] == NODE_TYPE_TRUTH_SHOWER:
+    #                     # Free node was a truth node
+    #                     matched_full_graph.add_edge(next(iter(p[0])), free_nodes[c], attached_in_pass=npass)
+    #                 else:
+    #                     # Free node was a pred node
+    #                     matched_full_graph.add_edge(next(iter(p[1])), free_nodes[c], attached_in_pass=npass)
+    #         graphs.append(matched_full_graph.copy())
+    #
+    #
+    #     self.calculated_graph = self._reduce_graph(matched_full_graph)
```

Comment:
If the above is not needed, can we remove it for now?
(as long as it is in the history)